src/daemon/socket.rs

```diff
@@ -2,22 +2,38 @@ use crate::{control_protocol::SendMessageArgs, daemon::DaemonContext};
 use bytes::Bytes;
 use earendil_crypt::Fingerprint;
 use earendil_packet::{Dock, Message};
+use rand::Rng;
+use serde::{Deserialize, Serialize};
 use smol::channel::Receiver;
 
+#[derive(Clone)]
```

Comment:
Deriving Clone here is subtly wrong due to a "double free" of the table entry in the Drop implementation. If you clone `socket` into `socket1` and `socket2`, and drop `socket1`, `socket2` will suddenly be invalid.

Either just don't implement Clone here and have users use Arc, or use some sort of more complicated structure on the inside which contains an Arc that contains an item that when dropped, deletes the corresponding entry in the table.